en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/data-compression.md

```diff
@@ -0,0 +1,176 @@
+    </tr>
+    <tr>
+        <td><code>contentType</code></td>
+        <td>Content types that should consider to apply data compression</td>
+    </tr>
+    <tr>
+        <td><code>enableForEtagHeader</code></td>
+        <td>Applies data compression even there is an ETag header</td>
+    </tr>
+    <tr>
+        <td colspan="2"><center><code><b>Compression Library relevant configurations</b></code></center></td>
+    </tr>
+    <tr>
+        <td><code>memoryLevel</code></td>
+        <td>
+            Amount of internal memory used by the gzip zlib library. Large values use more memory. The relevant value range
+            is 1 to 9.
+        </td>
+    </tr>
+    <tr>
+        <td><code>windowBits</code></td>
+        <td>
+            Represents the base two logarithmic of the compressor’s window size. Value range is from 9 to 15. Large window
+            sizes provide high compression by using more memory.
+        </td>
+    </tr>
+    <tr>
+        <td><code>compressionLevel</code></td>
+        <td>
+            Denotes zlib library's compression level. This attributes impacts speed and compression amount applied to the 
+            content. Value range is from 1 to 9. Level 9 provides the highest compression.
+        </td>
+    </tr>
+    <tr>
+        <td><code>compressionStrategy</code></td>
+        <td>
+            Denotes zlib library's compression strategy. Value can change based on the content type shared with the request. 
+            For most of the cases default strategy is the best choice.
+        </td>
+    </tr>
+    <tr>
+        <td><code>chunkSize</code></td>
+        <td>
+            Value for the zlib library's next output buffer. Denotes the amount in bytes.
+        </td>
+    </tr>
+</table>
+
+## Examples for data compression in Choreo Connect
+
+### Applying data compression for the request flow
+
+If `router.filters.compression.requestDirection` enable value assigned as `true`, Choreo Connect will compress the HTTP request data
+sent to the upstream.
+
+!!! note
+- To apply the data compression, content size should be higher than the value defined in the configurations (`minimumContentLength`).
+- Data compression will be applied only to the content types defined in the configurations.
+- If `Content-encoding` header is given in the request, Choreo Connect will not apply data compression during the request flow.
+
+ ``` bash tab="Format"
+ curl -X 'POST' "https://<CHOREO-CONNECT_ROUTER_HOST>:<CHOREO-CONNECT_ROUTER_PORT>/<API-context>/<API-resource>" -H "Content-type:application/json" -H "Authorization:Bearer $TOKEN" -H "Accept-encoding: gzip" --data <data> -k
+ ```
+
+ ``` bash tab="Example"
+ curl -X 'POST' "https://localhost:9095/v2/testAPI/*" -H "Content-type:application/json" -H "Accept-encoding: gzip" -H "Authorization:Bearer $TOKEN" -k --data '{"name":"John","age":30,"Address":"No 123, Street Name, Province, Country"}'
+ ```
+
+### Applying data compression for the response flow
```

Comment:
This feels slightly repetitive to me. We have almost the same text for both requests and responses. Can we merge both into one topic?